src/Library/Platform/Interface/PlatformEvents.h

```diff
@@ -72,3 +72,9 @@ class PlatformNativeEvent: public PlatformEvent {
  public:
     const void *nativeEvent = nullptr; // Pointer to a native event, in our case this is `SDL_Event`. Never `nullptr`.
 };
+
+class PlatformTextInputEvent : public PlatformEvent {
+ public:
+    PlatformWindow *window = nullptr;
+    const char *text = nullptr;
```

Comment:
Use `std::string`, this field should hold the text. Thanks to SSO it won't allocate in pretty much all cases.